File: app/src/main/java/com/sch/playandroid/test/TestActivity.java
package com.sch.playandroid.test;

import android.animation.ValueAnimator;
import android.os.Bundle;
import android.view.animation.DecelerateInterpolator;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.sch.playandroid.R;

/**
 * Created by Sch.
 * Date: 2020/4/27
 * description:
 */
public class TestActivity extends AppCompatActivity {
    private TextView tvNum;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        tvNum = findViewById(R.id.tvNum);
        tvNum.post(new Runnable() {
            @Override
            public void run() {
                ValueAnimator valueAnimator = new ValueAnimator();
                valueAnimator.setIntValues(0, 100);
                valueAnimator.setDuration(5000);
                valueAnimator.setInterpolator(new DecelerateInterpolator());
                valueAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
                    @Override
                    public void onAnimationUpdate(ValueAnimator valueAnimator) {
                        tvNum.setText("" + valueAnimator.getAnimatedValue());
                    }
                });
                valueAnimator.start();
            }
        });

    }
}

File: app/src/main/java/com/sch/playandroid/test/TestActivity.kt
package com.sch.playandroid.test

import android.os.Bundle
import android.util.Log
import com.sch.playandroid.R
import com.sch.playandroid.base.BaseActivity

/**
 * Created by Sch.
 * Date: 2020/5/6
 * description:
 */
class TestActivity : BaseActivity<TestContract.ITestPresenter>(), TestContract.ITestView {

    override fun getLayoutId(): Int {
        return R.layout.activity_test
    }

    override fun init(savedInstanceState: Bundle?) {
        mPresenter?.doSomeThings()
    }

    override fun onSuccess() {
        Log.e("test", "success")
    }

    override fun onError(ex: String) {
    }

    override fun createPresenter(): TestContract.ITestPresenter? {
        return TestPresenterImpl()
    }

}